pkg/kapp/config/config.go

```diff
@@ -53,6 +54,18 @@ type WaitRuleConditionMatcher struct {
 	Success bool
 }
 
+type WaitRuleYtt struct {
+	// Contracts are named and versioned (eg v1)
+	// to provide a stable interface to rule authors.
+	// Multiple contracts will be offered at the same time
+	// so that existing rules do not not break as we decide to evolve running environment.
+	FuncContractV1 *funcContractV1 `json:"funcContractV1"`
+}
+
+type funcContractV1 struct {
+	RulesStar string `json:"rules.star"`
```

Comment:
im thinking we should rename this field to `resource.star` and function should be called `is_done(resource)`

Reply:
Adopted.